src/persist/src/azure.rs

```diff
@@ -216,35 +200,97 @@ impl Blob for AzureBlob {
             // buffer into lgalloc.
             let mut buffer = match content_length {
                 1.. => {
-                    let region = self
-                        .metrics
+                    let region = metrics
                         .lgbytes
                         .persist_azure
                         .new_region(usize::cast_from(content_length));
                     PreSizedBuffer::Sized(region)
                 }
-                0 => PreSizedBuffer::Unknown(Vec::new()),
+                0 => {
+                    metrics.get_invalid_resp.inc();
+                    PreSizedBuffer::Unknown(SegmentedBytes::new())
+                }
             };
 
             let mut body = response.data;
             while let Some(value) = body.next().await {
                 let value = value.map_err(|e| {
                     ExternalError::from(anyhow!("Azure blob get body error: {}", e))
                 })?;
-                buffer.extend_from_slice(&value);
+
+                match &mut buffer {
+                    PreSizedBuffer::Sized(region) => region.extend_from_slice(&value),
+                    PreSizedBuffer::Unknown(segments) => segments.push(value),
+                }
             }
 
             // Spill our bytes to lgalloc, if they aren't already.
-            let lg_bytes = match buffer {
+            let lgbytes = match buffer {
                 PreSizedBuffer::Sized(region) => LgBytes::from(Arc::new(region)),
-                PreSizedBuffer::Unknown(buffer) => {
-                    self.metrics.lgbytes.persist_azure.try_mmap(buffer)
+                // Now that we've collected all of the segments, we know the size of our region.
+                PreSizedBuffer::Unknown(segments) => {
+                    let mut region = metrics.lgbytes.persist_azure.new_region(segments.len());
+                    for segment in segments.into_segments() {
+                        region.extend_from_slice(segment.as_ref());
+                    }
+                    LgBytes::from(Arc::new(region))
+                }
+            };
+
+            Ok(MaybeLgBytes::LgBytes(lgbytes))
+        }
+
+        let mut handles = Vec::new();
+        // TODO: the default chunk size is 1MB. We have not tried tuning it,
+        // but making this configurable / running some benchmarks could be
+        // valuable.
+        let mut stream = blob.get().into_stream();
+
+        while let Some(value) = stream.next().await {
```

Comment:
Could we `map`/`buffer_unordered` here instead of spinning up individual tasks? A bit closer to the S3 impl (which does not fork off individual tasks) and makes it easier to cap the concurrency per fetch...

Reply:
good call! refactored to use `FuturesOrdered` like the S3 blob impl